src/base1/cockpit.js

```diff
@@ -3244,7 +3244,7 @@ function factory() {
 
         function on_close(event, options) {
             dbus_debug("dbus close:", options);
-            channel.removeEventListener("ready", on_ready);
+            channel.removeEventListener("control", on_ready);
```

Comment:
I believe this should go the other way around. The ```channel.addEventListener()... above should connect to ```"ready"``` ... either that or the ```on_ready()``` function should check ```message["command"]``` and only handle ```"ready``` messages. But the former is clearer.